File: UnitTests/Samples/LateBreaking/DataGridViewCellIMEMode/VB/DataGridViewCellIMEMode/DataGridViewCellIMEMode.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------
Public Class DataGridViewCellIMEMode

    Private Sub DataGridView1_EditingControlShowing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.DataGridViewEditingControlShowingEventArgs) Handles DataGridView1.EditingControlShowing
        Select Case DataGridView1.CurrentCell.ColumnIndex
            Case 0
                e.Control.ImeMode = Windows.Forms.ImeMode.Hiragana
            Case 1
                e.Control.ImeMode = Windows.Forms.ImeMode.Katakana
            Case Else
                e.Control.ImeMode = Windows.Forms.ImeMode.Off
        End Select
    End Sub
End Class

File: UnitTests/Samples/LateBreaking/DataGridViewCellIMEMode/CS/DataGridViewCellIMEMode/DataGridViewCellIMEMode.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace Microsoft.Samples.DataGridViewCellIMEMode
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void dataGridView1_EditingControlShowing(object sender, DataGridViewEditingControlShowingEventArgs e)
        {
            switch (dataGridView1.CurrentCell.ColumnIndex)
            {
                case 0:
                    e.Control.ImeMode = ImeMode.Hiragana;
                    break;
                case 1:
                    e.Control.ImeMode = ImeMode.Katakana;
                    break;
                default:
                    e.Control.ImeMode = ImeMode.Off;
                    break;
            }
        }
    }
}